app/controllers/devise_token_auth/passwords_controller.rb

```diff
@@ -188,5 +182,13 @@ def with_reset_password_token token
     def render_not_found_error
       render_error(404, I18n.t('devise_token_auth.passwords.user_not_found', email: @email))
     end
+
+    def redirect_url
```

Comment:
It looks not bad.
I think you may rewrite it something like this:
```ruby


    before_action :validate_redirect_url_param, only: [:create, :edit]

    private

    def validate_redirect_url_param
      # give redirect value from params priority
      @redirect_url ||= params.fetch(
        :redirect_url,
        DeviseTokenAuth.default_password_reset_url
      )

      render_create_error_missing_redirect_url unless @redirect_url
      render_create_error_not_allowed_redirect_url if blacklisted_redirect_url?
    end

    def redirect_url
      @redirect_url
    end
```
By this approach you can cover two actions at the same code. I suppose edit action needs the same `redirect_url` validation.